pdlua.c

```diff
@@ -1108,43 +1144,67 @@ static t_int *pdlua_perform(t_int *w){
         }
         lua_pop(__L(), 1);
     }
-
     lua_pop(__L(), 1); /* pop the global "pd" */
-    
+
     PDLUA_DEBUG("pdlua_perform: end. stack top %d", lua_gettop(__L()));
-    
-    return w + o->siginlets + o->sigoutlets + 3;
+
+    return w + 2 + o->siginlets + o->sigoutlets;
 }
 
-static void pdlua_dsp(t_pdlua *x, t_signal **sp){
+static void pdlua_dsp(t_pdlua *x, t_signal **sp) {
     int sum = x->siginlets + x->sigoutlets;
     if(sum == 0) return;
     x->sig_warned = 0;
-    
+
     PDLUA_DEBUG("pdlua_dsp: stack top %d", lua_gettop(__L()));
+
+    x->sp = sp; // FIXME: is this the way? (setting this for signal_setmultiout)
```

Comment:
i assume that this is fine since i don't see another simple way? ... the signal pointer is stored here only for `signal_setmultiout` to access it then.

Reply:
closing this comment since this approach is probably not wrong.